content/articles/image-processing-with-coil-in-android/index.md

```diff
@@ -0,0 +1,486 @@
+    val link = ImageLinks().randomLink()
+    binding.imageView.apply {
+        scaleType = ImageView.ScaleType.CENTER_CROP
+        load(link) {
+            size(300, 300)
+        }
+    }
+}
+```
+
+#### vii). Combined Transformations
+In some cases, you may want to apply more than just one transformation on the same image at the same time. This can be achieved as shown below:
+
+```kotlin
+binding.combined.setOnClickListener {
+    val link = ImageLinks().randomLink()
+    binding.imageView.load(link) {
+        crossfade(750)
+        transformations(
+            CircleCropTransformation(),
+            GrayscaleTransformation() // e.t.c
+        )
+        build()
+    }
+    updateState("Combined State")
+}
+```
+Notice the use of commas to separate transformations.
+
+### Cancelling Requests
+In some cases, a network request needs to be canceled, especially if it takes too long to respond. Request cancellation is also referred to as disposal, as shown in the snippet below.
+
+```kotlin
+imageView.load(link){
+        ...
+    }.dispose()
+```
+This frees resources associated with the request by canceling any work in progress. This method is [idempotent](https://en.wikipedia.org/wiki/Idempotence) meaning that multiple applications do not affect the result of the first app.
+
+### Memory Caching
+Caching makes it possible to display remote images even when the internet connection is lost. For this to happen, the image must have been fetched in at least one instance.
+
+Coil makes use of `OkHttp` which is an `HTTP client` that supports the `SPDY` protocol. Refer to this [official documentation](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-cache/) for further reading.
+
+### Exception handling
+Keeping track of a request is important as it helps in handling errors encountered during the process. 
+
+Coil uses a `listener()` function that accepts two lambdas, `onSuccess` and `onError`, as demonstrated below: 
+
+```kotlin
+// handle error button click
+binding.error.setOnClickListener {
+    binding.imageView.load("https://a/bad/link") {
+        listener(
+            // pass two arguments
+            onSuccess = { _, _ ->
+                Toast.makeText(this@MainActivity, "Success", Toast.LENGTH_SHORT).show()
+            },
+            onError = { request: ImageRequest, throwable: Throwable ->
+                request.error
+                Toast.makeText(this@MainActivity, "$throwable", Toast.LENGTH_SHORT).show()
+            })
+        // setup error image
+        error(R.drawable.ic_error_image)
+        
+    }
+}
+```
+An error image is used to indicate that something went wrong when resolving the request.
+
+### Conclusion
+Congratulations! you can now comfortably use `Coil` to process images in Android. You can use this knowledge to build more powerful applications. The source code for this tutorial can be found [on this Github repository](https://github.com/nonimdiana/image-processing-with-coil).
```

Comment:
Hello @WanjaMIKE, I'm glad you noticed this. I've fixed it and everything is now working!